src/server/com/yetanalytics/datasim/server.clj

```diff
@@ -240,29 +241,48 @@
 ;; Routes and server configs
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(def routes
-  (route/expand-routes
-   #{["/health"
-      :get        health
-      :route-name :datasim.route/health]
-     ["/api/v1/generate"
-      :post (into common-interceptors
-                  [(multipart-params)
-                   generate])]
-     ["/api/v1/download-url"
-      :get  (into common-interceptors
-                  [download-url])]}))
-
 (defn create-server
   []
-  (http/create-server
-   {::http/routes          routes
-    ::http/type            :jetty
-    ::http/allowed-origins ["https://yetanalytics.github.io"
-                            "http://localhost:9091"]
-    ::http/host            "0.0.0.0"
-    ::http/port            9090
-    ::http/join?           false}))
+  (let [root-path       (get env :api-root-path "")
+        _               (when (not-empty root-path)
+                          (when-not (str/starts-with? root-path "/")
+                            (throw (ex-info "API_ROOT_PATH must start with /"
+                                            {:type      ::invalid-root-path
+                                             :root-path root-path})))
+                          (when (str/ends-with? root-path "/")
+                            (throw (ex-info "API_ROOT_PATH must not end with /"
+                                            {:type      ::invalid-root-path
+                                             :root-path root-path}))))
```

Comment:
Probably worth separating this out into its own assert function.

Reply:
https://github.com/yetanalytics/datasim/pull/72/commits/7ffb0016330611f9374fc18ab48e881db9cda93a